internal/guest/runtime/hcsv2/uvm.go

```diff
@@ -292,6 +294,66 @@ func setupSandboxHugePageMountsPath(id string) error {
 	return storage.MountRShared(mountPath)
 }
 
+func getUserNameForID(spec *specs.Spec, uid uint32) (string, error) {
+	users, err := user.ParsePasswdFileFilter(filepath.Join(spec.Root.Path, "/etc/passwd"), func(user user.User) bool {
+		return uint32(user.Uid) == uid
+	})
+	if err != nil {
+		return "", err
+	}
+
+	if len(users) != 1 {
+		return "", errors.Errorf("expected exactly 1 user matched '%d'", len(users))
+	}
+	return users[0].Name, nil
+}
+
+func getGroupNameForID(spec *specs.Spec, gid uint32) (string, error) {
```

Comment:
same here, we have code under `/internal/guest/runtime/hcsv2/spec.go`

Reply:
Removed code duplication